src/js/components/releases/releases.js

```diff
@@ -0,0 +1,323 @@
+import AddArtifactDialog from './dialogs/addartifact';
+import ReleaseDetails from './releasedetails';
+import ReleasesList from './releaseslist';
+
+const refreshArtifactsLength = 60000;
+
+const tabs = [
+  { key: 'releases', title: 'Releases', component: ReleasesList, isApplicable: () => true },
+  {
+    key: 'delta',
+    title: 'Delta Artifacts generation',
+    component: 'div',
+    isApplicable: ({ features: { hasDeltaProgress }, tenantCapabilities: { canDelta } }) => hasDeltaProgress && canDelta
+  }
+];
+
+const EmptyState = ({ canUpload, dropzoneRef, uploading, onDrop, onUpload }) => (
+  <div className="dashboard-placeholder fadeIn" style={{ fontSize: '16px', margin: '8vh auto' }} ref={dropzoneRef}>
+    <Dropzone activeClassName="active" disabled={uploading} multiple={false} noClick={true} onDrop={onDrop} rejectClassName="active">
+      {({ getRootProps, getInputProps }) => (
+        <div {...getRootProps({ className: uploading ? 'dropzone disabled muted' : 'dropzone' })} onClick={() => onUpload()}>
+          <input {...getInputProps()} disabled={uploading} />
+          <p>
+            There are no Releases yet.{' '}
+            {canUpload && (
+              <>
+                <a>Upload an Artifact</a> to create a new Release
+              </>
+            )}
+          </p>
+        </div>
+      )}
+    </Dropzone>
+  </div>
+);
+
+const Header = ({
+  canUpload,
+  existingTags = [],
+  features,
+  hasReleases,
+  releasesListState,
+  setReleasesListState,
+  onUploadClick,
+  tenantCapabilities,
+  uploadButtonRef
+}) => {
+  const { hasReleaseTags } = features;
+  const { selectedTags = [], searchTerm, searchTotal, tab = tabs[0].key, total } = releasesListState;
+
+  const searchUpdated = searchTerm => setReleasesListState({ searchTerm });
+
+  const onTabChanged = (e, tab) => setReleasesListState({ tab });
+
+  const onTagSelectionChanged = ({ selection }) => setReleasesListState({ selectedTags: selection });
+
+  const availableTabs = useMemo(
+    () =>
+      tabs.reduce((accu, tab) => {
+        if (tab.isApplicable({ features, tenantCapabilities })) {
+          accu.push(tab);
+        } else {
+          accu.push({ ...tab, disabled: true });
+        }
+        return accu;
+      }, []),
+    [JSON.stringify(tenantCapabilities)]
+  );
+
+  return (
+    <div>
+      <div className="flexbox space-between">
+        <Tabs value={tab} onChange={onTabChanged} textColor="primary" style={{ alignSelf: 'flex-start' }}>
+          {availableTabs.map(({ disabled = false, key, title }) => (
+            <Tab key={key} label={title} value={key} disabled={disabled} />
+          ))}
+        </Tabs>
+        <div>
+          {canUpload && (
+            <>
+              <Button
+                ref={uploadButtonRef}
+                color="secondary"
+                onClick={onUploadClick}
+                startIcon={<CloudUpload fontSize="small" />}
+                style={{ marginTop: 30, minWidth: 164 }}
```

Comment:
Can we remove `marginTop: 30`? It messes up the layout a bit

Reply:
with a `marginTop: 12` it aligned with the tab underline - would that also be ok?

Reply:
Perfect!